crates/keth-run/src/main.rs

```diff
@@ -0,0 +1,102 @@
+
+    /// Path to compiled Cairo program
+    #[arg(long, value_name = "FILE", default_value = "build/main_compiled.json")]
+    compiled_program: PathBuf,
+
+    /// Generate Stwo proof instead of prover inputs
+    #[arg(long)]
+    stwo_proof: bool,
+
+    /// Path to save the Stwo proof (required when --stwo-proof is used)
+    #[arg(long, value_name = "FILE", required_if_eq("stwo_proof", "true"))]
+    proof_path: Option<PathBuf>,
+
+    /// Verify the Stwo proof after generation (only used with --stwo-proof)
+    #[arg(long)]
+    verify: bool,
+}
+
+// Check if the PYTHONPATH, PYTHONHOME, and PYO3_PYTHON variables are set, panics if any one is
+// missing
+fn check_env() {
+    dotenvy::dotenv().ok();
+
+    dotenvy::var("PYTHONPATH").expect("PYTHONPATH environment variable not set");
+    dotenvy::var("PYTHONHOME").expect("PYTHONHOME environment variable not set");
+    dotenvy::var("PYO3_PYTHON").expect("PYO3_PYTHON environment variable not set");
+
+    println!("PYTHONPATH: {}", dotenvy::var("PYTHONPATH").unwrap());
+    println!("PYTHONHOME: {}", dotenvy::var("PYTHONHOME").unwrap());
+    println!("PYO3_PYTHON: {}", dotenvy::var("PYO3_PYTHON").unwrap());
+
+    println!("cargo:rerun-if-env-changed=PYTHONPATH");
+    println!("cargo:rerun-if-env-changed=PYTHONHOME");
+    println!("cargo:rerun-if-env-changed=PYO3_PYTHON");
+}
+
+fn main() -> Result<()> {
+    check_env();
+
+    let args = Args::parse();
+
+    if !args.compiled_program.exists() {
+        anyhow::bail!("Compiled program not found: {}", args.compiled_program.display());
+    }
+
+    let zkpi_path = args.data_dir.join(format!("{}.json", args.block_number));
+    if !zkpi_path.exists() {
+        anyhow::bail!("ZKPI data not found: {}", zkpi_path.display());
+    }
+
+    if !args.output_dir.exists() {
+        std::fs::create_dir_all(&args.output_dir)
+            .map_err(|e| anyhow::anyhow!("Error creating output directory: {e}"))?;
+    }
+
+    let _ = Python::with_gil(|py| -> Result<()> {
+        // Get the current directory and add it to Python's path
+        let current_dir = std::env::current_dir()?;
+        // Add the cairo directory to Python's path
+        let sys = py.import("sys")?;
+        let path = sys.getattr("path")?;
+        path.call_method1("append", (current_dir.to_str().unwrap(),))?;
+
+        let prove_block_module = py.import("cairo.scripts.prove_block")?;
+        let load_zkpi_fixture = prove_block_module.getattr("load_zkpi_fixture")?;
+        let program_inputs = load_zkpi_fixture.call1((zkpi_path.to_str().unwrap(),))?;
+
+        let run_proof_mode = prove_block_module.getattr("run_proof_mode")?;
+        run_proof_mode.call1((
+            "main",
+            program_inputs,
+            args.compiled_program.to_str().unwrap(),
+            args.output_dir.to_str().unwrap(),
+            args.stwo_proof,
+            args.proof_path.as_ref().map(|p| p.to_str().unwrap()),
+            args.verify,
+        ))?;
+
+        Ok(())
```

Comment:
useless?

Reply:
the result types are different between `run_proof_mode` (returns `Result<pyo3::Bound<'_, PyAny>, _>`)
and what the closure expects `Result<(), _>` so I don't think there is a better way